In [5]:
import pandas as pd
import os

In [3]:
df = pd.read_csv("/Users/beneck/Library/CloudStorage/OneDrive-NortheasternUniversity/Boise Project/Data/SCAN Data Usage Experiment/2024-07-17_00-03-58.fp", sep='\t', header=1).drop('Status_0', axis=1)
df

Date/Time  200.00  202.50  205.00  207.50  210.00  212.50  \
0    2024.07.17  00:03:58     NaN     NaN     NaN     NaN     NaN     NaN   
1    2024.07.17  00:18:58     NaN     NaN     NaN     NaN     NaN     NaN   
2    2024.07.17  00:33:58     NaN     NaN     NaN     NaN     NaN     NaN   
3    2024.07.17  00:48:58     NaN     NaN     NaN     NaN     NaN     NaN   
4    2024.07.17  01:03:58     NaN     NaN     NaN     NaN     NaN     NaN   
..                    ...     ...     ...     ...     ...     ...     ...   
814  2024.07.25  11:33:58     NaN     NaN     NaN     NaN     NaN     NaN   
815  2024.07.25  11:48:58     NaN     NaN     NaN     NaN     NaN     NaN   
816  2024.07.25  12:03:58     NaN     NaN     NaN     NaN     NaN     NaN   
817  2024.07.25  12:18:58     NaN     NaN     NaN     NaN     NaN     NaN   
818  2024.07.25  12:33:58     NaN     NaN     NaN     NaN     NaN     NaN   

     215.00  217.50  220.00  ...  727.50  730.00  732.50  735.00  737.50  \
0       NaN     NaN -17.016  ...  -8.078  -8.047  -8.094  -8.250  -8.406   
1       NaN     NaN -16.922  ...  -7.906  -8.031  -8.094  -8.219  -8.359   
2       NaN     NaN -16.875  ...  -7.797  -7.984  -8.141  -8.250  -8.328   
3       NaN     NaN -16.875  ...  -7.844  -7.984  -8.047  -8.203  -8.344   
4       NaN     NaN -16.797  ...  -7.875  -8.062  -8.141  -8.219  -8.359   
..      ...     ...     ...  ...     ...     ...     ...     ...     ...   
814     NaN     NaN -17.234  ...  -8.031  -8.031  -8.094  -8.328  -8.562   
815     NaN     NaN -17.125  ...  -7.891  -7.906  -7.906  -8.094  -8.406   
816     NaN     NaN -17.281  ...  -8.031  -8.078  -8.156  -8.328  -8.531   
817     NaN     NaN -17.109  ...  -7.828  -7.828  -7.922  -8.125  -8.359   
818     NaN     NaN -17.078  ...  -7.937  -7.922  -7.922  -8.031  -8.266   

     740.00  742.50  745.00  747.50  750.00  
0       NaN     NaN     NaN     NaN     NaN  
1       NaN     NaN     NaN     NaN     NaN  
2       NaN     NaN     NaN     NaN     NaN  
3       NaN     NaN     NaN     NaN     NaN  
4       NaN     NaN     NaN     NaN     NaN  
..      ...     ...     ...     ...     ...  
814     NaN     NaN     NaN     NaN     NaN  
815     NaN     NaN     NaN     NaN     NaN  
816     NaN     NaN     NaN     NaN     NaN  
817     NaN     NaN     NaN     NaN     NaN  
818     NaN     NaN     NaN     NaN     NaN  

[819 rows x 222 columns]

In [7]:
filesize= os.path.getsize("/Users/beneck/Library/CloudStorage/OneDrive-NortheasternUniversity/Boise Project/Data/SCAN Data Usage Experiment/2024-07-17_00-03-58.fp")
filesize

1303960

In [11]:
kb_per_row = filesize/1000/df.index.size
kb_per_row

1.5921367521367522

In [12]:
max_filesize = 2000

data_points_for_maxfile = max_filesize/kb_per_row
data_points_for_maxfile

1256.1735022546704

# 30s logging interval, ran for just under 24h
- max filesize upped to 4MB from 2MB on S::CAN

In [35]:
df2 = pd.read_csv("/Users/beneck/Library/CloudStorage/OneDrive-NortheasternUniversity/Boise Project/Data/SCAN Data Usage Experiment/2024-07-31_00-03-58.fp", sep='\t', header=1).drop('Status_0', axis=1)
df2

Date/Time  200.00  202.50  205.00  207.50  210.00  212.50  \
0     2024.07.17  00:03:58     NaN     NaN     NaN     NaN     NaN     NaN   
1     2024.07.17  00:18:58     NaN     NaN     NaN     NaN     NaN     NaN   
2     2024.07.17  00:33:58     NaN     NaN     NaN     NaN     NaN     NaN   
3     2024.07.17  00:48:58     NaN     NaN     NaN     NaN     NaN     NaN   
4     2024.07.17  01:03:58     NaN     NaN     NaN     NaN     NaN     NaN   
...                    ...     ...     ...     ...     ...     ...     ...   
1515  2024.07.31  01:30:23     NaN     NaN     NaN     NaN     NaN     NaN   
1516  2024.07.31  01:31:23     NaN     NaN     NaN     NaN     NaN     NaN   
1517  2024.07.31  01:32:23     NaN     NaN     NaN     NaN     NaN     NaN   
1518  2024.07.31  01:33:23     NaN     NaN     NaN     NaN     NaN     NaN   
1519  2024.07.31  01:34:23     NaN     NaN     NaN     NaN     NaN     NaN   

      215.00  217.50  220.00  ...  727.50  730.00  732.50  735.00  737.50  \
0        NaN     NaN -17.016  ...  -8.078  -8.047  -8.094  -8.250  -8.406   
1        NaN     NaN -16.922  ...  -7.906  -8.031  -8.094  -8.219  -8.359   
2        NaN     NaN -16.875  ...  -7.797  -7.984  -8.141  -8.250  -8.328   
3        NaN     NaN -16.875  ...  -7.844  -7.984  -8.047  -8.203  -8.344   
4        NaN     NaN -16.797  ...  -7.875  -8.062  -8.141  -8.219  -8.359   
...      ...     ...     ...  ...     ...     ...     ...     ...     ...   
1515     NaN     NaN -17.937  ...  -8.062  -8.094  -8.141  -8.328  -8.453   
1516     NaN     NaN -17.703  ...  -7.922  -8.000  -8.062  -8.203  -8.359   
1517     NaN     NaN -17.562  ...  -7.891  -7.937  -8.000  -8.187  -8.344   
1518     NaN     NaN -17.516  ...  -7.781  -7.844  -7.953  -8.109  -8.125   
1519     NaN     NaN -17.734  ...  -7.875  -7.953  -8.047  -8.219  -8.312   

      740.00  742.50  745.00  747.50  750.00  
0        NaN     NaN     NaN     NaN     NaN  
1        NaN     NaN     NaN     NaN     NaN  
2        NaN     NaN     NaN     NaN     NaN  
3        NaN     NaN     NaN     NaN     NaN  
4        NaN     NaN     NaN     NaN     NaN  
...      ...     ...     ...     ...     ...  
1515     NaN     NaN     NaN     NaN     NaN  
1516     NaN     NaN     NaN     NaN     NaN  
1517     NaN     NaN     NaN     NaN     NaN  
1518     NaN     NaN     NaN     NaN     NaN  
1519     NaN     NaN     NaN     NaN     NaN  

[1520 rows x 222 columns]

In [36]:
df2['Timestamp'] = pd.to_datetime(df2['Date/Time'])
df2.set_index('Timestamp', inplace=True)

In [38]:
df2.loc['07-30-2024 00:00':'07-31-2024 23:59']

Date/Time  200.00  202.50  205.00  207.50  \
Timestamp                                                                   
2024-07-30 14:13:06  2024.07.30  14:13:06     NaN     NaN     NaN     NaN   
2024-07-30 14:18:53  2024.07.30  14:18:53     NaN     NaN     NaN     NaN   
2024-07-30 14:19:53  2024.07.30  14:19:53     NaN     NaN     NaN     NaN   
2024-07-30 14:20:53  2024.07.30  14:20:53     NaN     NaN     NaN     NaN   
2024-07-30 14:21:53  2024.07.30  14:21:53     NaN     NaN     NaN     NaN   
...                                   ...     ...     ...     ...     ...   
2024-07-31 01:30:23  2024.07.31  01:30:23     NaN     NaN     NaN     NaN   
2024-07-31 01:31:23  2024.07.31  01:31:23     NaN     NaN     NaN     NaN   
2024-07-31 01:32:23  2024.07.31  01:32:23     NaN     NaN     NaN     NaN   
2024-07-31 01:33:23  2024.07.31  01:33:23     NaN     NaN     NaN     NaN   
2024-07-31 01:34:23  2024.07.31  01:34:23     NaN     NaN     NaN     NaN   

                     210.00  212.50  215.00  217.50  220.00  ...  727.50  \
Timestamp                                                    ...           
2024-07-30 14:13:06     NaN     NaN     NaN     NaN -17.031  ...  -7.969   
2024-07-30 14:18:53     NaN     NaN     NaN     NaN -16.859  ...  -7.859   
2024-07-30 14:19:53     NaN     NaN     NaN     NaN -16.984  ...  -7.906   
2024-07-30 14:20:53     NaN     NaN     NaN     NaN -16.937  ...  -7.953   
2024-07-30 14:21:53     NaN     NaN     NaN     NaN -16.953  ...  -7.937   
...                     ...     ...     ...     ...     ...  ...     ...   
2024-07-31 01:30:23     NaN     NaN     NaN     NaN -17.937  ...  -8.062   
2024-07-31 01:31:23     NaN     NaN     NaN     NaN -17.703  ...  -7.922   
2024-07-31 01:32:23     NaN     NaN     NaN     NaN -17.562  ...  -7.891   
2024-07-31 01:33:23     NaN     NaN     NaN     NaN -17.516  ...  -7.781   
2024-07-31 01:34:23     NaN     NaN     NaN     NaN -17.734  ...  -7.875   

                     730.00  732.50  735.00  737.50  740.00  742.50  745.00  \
Timestamp                                                                     
2024-07-30 14:13:06  -7.969  -7.984  -8.078  -8.281     NaN     NaN     NaN   
2024-07-30 14:18:53  -7.875  -7.937  -7.969  -8.078     NaN     NaN     NaN   
2024-07-30 14:19:53  -7.984  -8.094  -8.156  -8.234     NaN     NaN     NaN   
2024-07-30 14:20:53  -7.953  -8.031  -8.172  -8.312     NaN     NaN     NaN   
2024-07-30 14:21:53  -7.937  -8.078  -8.156  -8.297     NaN     NaN     NaN   
...                     ...     ...     ...     ...     ...     ...     ...   
2024-07-31 01:30:23  -8.094  -8.141  -8.328  -8.453     NaN     NaN     NaN   
2024-07-31 01:31:23  -8.000  -8.062  -8.203  -8.359     NaN     NaN     NaN   
2024-07-31 01:32:23  -7.937  -8.000  -8.187  -8.344     NaN     NaN     NaN   
2024-07-31 01:33:23  -7.844  -7.953  -8.109  -8.125     NaN     NaN     NaN   
2024-07-31 01:34:23  -7.953  -8.047  -8.219  -8.312     NaN     NaN     NaN   

                     747.50  750.00  
Timestamp                            
2024-07-30 14:13:06     NaN     NaN  
2024-07-30 14:18:53     NaN     NaN  
2024-07-30 14:19:53     NaN     NaN  
2024-07-30 14:20:53     NaN     NaN  
2024-07-30 14:21:53     NaN     NaN  
...                     ...     ...  
2024-07-31 01:30:23     NaN     NaN  
2024-07-31 01:31:23     NaN     NaN  
2024-07-31 01:32:23     NaN     NaN  
2024-07-31 01:33:23     NaN     NaN  
2024-07-31 01:34:23     NaN     NaN  

[684 rows x 222 columns]

In [14]:
filesize2= os.path.getsize("/Users/beneck/Library/CloudStorage/OneDrive-NortheasternUniversity/Boise Project/Data/SCAN Data Usage Experiment/2024-07-31_00-03-58.fp")
filesize2

2418905

In [15]:
kb_per_row2 = filesize2/1000/df2.index.size
kb_per_row2

1.5913848684210528

- ok, so seems like the limit is 1520 rows. 
- but the file didn't clear all the way - more than half of it is from the last experiment
- so maybe it is possible to up the limit but I'm just going to assume it isn't and set the mneter back to what it was